app/routes/deprecations.js

```diff
@@ -1,32 +1,39 @@
-import { set } from '@ember/object';
-import TabRoute from "ember-inspector/routes/tab";
+import { Promise } from 'rsvp';
+import { setProperties } from '@ember/object';
+import TabRoute from 'ember-inspector/routes/tab';
 
 export default TabRoute.extend({
-  setupController() {
-    let port = this.port;
-    port.on('deprecation:deprecationsAdded', this, this.deprecationsAdded);
-    port.send('deprecation:watch');
+  model() {
+    return new Promise(resolve => {
+      this.port.one('deprecation:deprecationsAdded', resolve);
+      this.port.send('deprecation:watch');
+    });
+  },
```

Comment:
This now blocks until we received the message from ember-debug, which is unconditional (it still sends it even when there are no deprecations).